### PR TITLE
Changelog for 0.5.0

### PR DESCRIPTION
**Pull request by rgrunber**
Signed-off-by: Roland Grunberg <rgrunber@redhat.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
 # MicroProfile Language Server Changelog
 
+## [0.5.0](https://github.com/eclipse/lsp4mp/milestone/5?closed=1) (July 28, 2022)
+### Enhancements
+ * performance - Delay revalidation and handle validation cancellation correctly. See [#252](https://github.com/eclipse/lsp4mp/pull/252).
+ * enhancement - Property file with property expressions (without default value) are flagged as wrong. See [#227](https://github.com/eclipse/lsp4mp/issues/227).
+
```
Comment (datho7561) [inline CHANGELOG.md:7]: PR 225/Issue 224

**Review by datho7561 — CHANGES_REQUESTED** [review]
Good start, most of my comments are just about formatting the Changelog to be consistent with the other entries. I think we had a few noteworthy changes that weren't tagged or didn't have related PRs, please let me know if I got anything wrong though.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,33 @@
 # MicroProfile Language Server Changelog
 
+## [0.5.0](https://github.com/eclipse/lsp4mp/milestone/5?closed=1) (July 28, 2022)
+### Enhancements
+ * performance - Delay revalidation and handle validation cancellation correctly. See [#252](https://github.com/eclipse/lsp4mp/pull/252).
+ * enhancement - Property file with property expressions (without default value) are flagged as wrong. See [#227](https://github.com/eclipse/lsp4mp/issues/227).
```
[the remark this answers is not in the document]
Comment (rgrunber) [reply CHANGELOG.md:6]: good catch :|

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,33 @@
...
+
+### Bug Fixes
+ * bug fix - Language Server attempts to calculate code actions for stale diagnostics. See [#272](https://github.com/eclipse/lsp4mp/issues/272).
+ * bug fix - Hovering property value fails with NPE. See [#265](https://github.com/eclipse/lsp4mp/issues/265).
+ * bug fix - Completing property name with existing value will replace current value with default value. See [#264](https://github.com/eclipse/lsp4mp/issues/264).
+ * bug fix - Empty completion when completion is triggered before the assign `=`. See [#255](https://github.com/eclipse/lsp4mp/issues/255).
+ * bug fix - Improve validation by handling some known corner cases. [#249](https://github.com/eclipse/lsp4mp/issues/249), [#235](https://github.com/eclipse/lsp4mp/issues/235), [#233](https://github.com/eclipse/lsp4mp/issues/233), [#232](https://github.com/eclipse/lsp4mp/issues/232), [#228](https://github.com/eclipse/lsp4mp/issues/228).
+
```
[the remark this answers is not in the document]
Comment (rgrunber) [reply CHANGELOG.md:14]: PR 260 & 261 are Issues #235 & #249 . I included them in the giant "Improve validation by handling some known corner cases" to keep things a bit more compact.

File: CHANGELOG.md (hunk 1 above)
Comment (rgrunber) [reply CHANGELOG.md:7]: I'll include it as part of the entry that 227 is on as they seemed closely related.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,33 @@
...
+### Other
+ * other - `javaASTValidator` schema refers to the wrong interface name. See [#234](https://github.com/eclipse/lsp4mp/issues/234).
+ * other - Use JDT-LS 1.13.0 and build with Java 17. See [#266](https://github.com/eclipse/lsp4mp/issues/266).
+ * other - Move to LSP4j 0.14.0. See [#254](https://github.com/eclipse/lsp4mp/issues/254).
+ * other - `PropertyReplacerStrategy` is in wrong package. See [#239](https://github.com/eclipse/lsp4mp/issues/239).
+ * other - Added `JDTTypeUtils.isVoidReturnType` for `QuarkusConfigMappingProvider` void check. See [#246](https://github.com/eclipse/lsp4mp/pull/246).
+ * other - Improved MicroProfile property value expression diagnostic message. See [#242](https://github.com/eclipse/lsp4mp/pull/242).
+ * other - Update quarkus-spring-web so that it is not vulnerable to CVE-2022-22965. See [#238](https://github.com/eclipse/lsp4mp/issues/238).
+ * other - Rename profile parameter to ordinal for better readability. See [#236](https://github.com/eclipse/lsp4mp/issues/236).
+ * other - `plugin.xml` not part of the test bundle. See [#230](https://github.com/eclipse/lsp4mp/issues/230).
```
[the remark this answers is not in the document]
Comment (rgrunber) [reply CHANGELOG.md]: I've moved a few. Note that I've kept some that while mentioning something is "wrong", do not fail in any way. If it's just a convenience I opted to leave it under other. Let me know if there's any others you think should be moved.

**Pull request merged**